Bump all matrix entries except authenticators to PHP 7.4

Cheddam opened the pull request:
RealMe is known not to be compatible with 7.2+ yet, so I've pinned the authenticator test suite to 7.1 for the time being.

dnsl48 commented on .travis.yml:
```diff
@@ -22,39 +22,39 @@ env:
       env: PHPUNIT_TEST=optional-authenticators
-    - php: 7.1
+    - php: 7.4snapshot
       env: PHPUNIT_TEST=mfa
```
```suggestion
```

Cheddam commented:
I've adjusted the PR to reflect the simplification you suggested, but kept 7.4snapshot so we can get another picture of what support is looking like after some other fixes to modules and the introduction of the `phpunit-mock-objects` fork to our dev dependencies. Marking the PR as WIP for now.

brynwhyman commented:
The RealMe module _should_ be PHP7.2 compatible now (at least from a syntax point of view).

We should pick this issue back up as a start to see what work is involved in getting PHP7.4 compatibility across the board.

Cheddam commented:
Update:

- Travis officially supports 7.4 stable, so I've updated the config to reflect that
- The builds are now passing
- RealMe still doesn't support 7.2+, so that's still pinned to 7.1 (won't be long now)
- I've squashed all of the commits down, so this is ready to merge
